### PR TITLE
update "First Steps" tutorial

### LINKED ISSUE #541: update first-steps documentation
Natalie has gone through the first-steps tutorial. These are her comments:

---

1. General note: maybe it would be nice to link to a general data science project pipeline tutorial of sorts at the beginning? Or some useful resources for setting up a general working environment?
2. Launch JupyterLab button in top right is missing (instructions are misleading)
3. No info about the (default) settings to launch JupyterLab is included in the instructions - maybe some recommended configurations depending on scope of project?
4. "Transition" loading page for JupyterLab is not so friendly (was wondering if it was loading or if it hung up?)
5. Small detail, but this instruction should have a comma after the first clause, otherwise kind of confusing: "To start working with the bicycle data, we have already created a sample notebook that does some data cleaning and visualization."
6. Before the section "Reuse your work", I got the message: "Error: The repository is dirty. Please use the "git" command to clean it." Maybe add instructions/tips to regularly commit and push changes to keep the repository clean?
7. A GNU screen comes up sometimes if you don't include message with git commit - maybe link to a tutorial for this?
8. After running "renku status", I get this message which differs from output instructions:
```
/home/jovyan/.local/pipx/venvs/renku/lib/python3.6/site-packages/renku/models/provenance/activities.py:597: YAMLLoadWarning:
  *** Calling yaml.load() without Loader=... is deprecated.
  *** The default Loader is unsafe.
  *** Please read https://msg.pyyaml.org/load for full details.
  process = CWLClass.from_cwl(yaml.load(data))
/home/jovyan/.local/pipx/venvs/renku/lib/python3.6/site-packages/renku/models/provenance/activities.py:597: YAMLLoadWarning:
  *** Calling yaml.load() without Loader=... is deprecated.
  *** The default Loader is unsafe.
  *** Please read https://msg.pyyaml.org/load for full details.
  process = CWLClass.from_cwl(yaml.load(data))
On branch master
All files were generated from the latest inputs.
```

### PR DESCRIPTION
**Pull request by lorenzo-cavazzi**
This refreshes the "First Steps" tutorial with a few fixes, updated screenshots and more verbose console outputs.

The first commit closes #541 -- although the first point will be addressed by a new tutorial and is not part of this PR.
I am not sure about the (few) changes in the second commit, I kept them separate so we can easily avoid to merge.

### CODE REVIEW

File: docs/user/firststeps.rst [hunk 1]
```diff
@@ -29,36 +30,58 @@ On renkulab.io_ you can sign in using with your GitHub
...
 
-To more easily find your project later, you can give it a star:
+.. image:: ../_static/images/ui_new_project.png
+    :width: 100%
+    :align: center
+    :alt: Create a new project
 
-.. image:: ../_static/images/ui_star_project.png
+Now that we have a project, we can start working on it by starting a
+new JupyterLab. Click on **Notebook Servers** (1), then on
```
Comment (ableuler) [inline docs/user/firststeps.rst]:
```suggestion
new JupyterLab notebook server. Click on **Notebook Servers** (1), then on
```

File: docs/user/firststeps.rst [hunk 2]
```diff
@@ -29,36 +30,58 @@ On renkulab.io_ you can sign in using with your GitHub
...
+    :alt: Head to Notebook ui_notebook_servers
+
+The default settings are fine, click on **Launch Server**. You will see
+a new row with a yellow button on the right. Wait until its color turns
+to green and the text from **Starting** to **Running**
+
+.. note::
+
+    Please be patient, the first time you start a server it may require
+    up to one minute
```
Comment (ableuler) [inline docs/user/firststeps.rst]:
```suggestion
    up to one minute.
```

File: docs/user/firststeps.rst [hunk 3]
```diff
@@ -29,36 +30,58 @@ On renkulab.io_ you can sign in using with your GitHub
...
+new JupyterLab. Click on **Notebook Servers** (1), then on
+**Start new server** (2).
+
+.. image:: ../_static/images/ui_notebook_servers.png
     :width: 100%
     :align: center
-    :alt: Star a project
+    :alt: Head to Notebook ui_notebook_servers
+
+The default settings are fine, click on **Launch Server**. You will see
```
Comment (ableuler) [inline docs/user/firststeps.rst]:
```suggestion
The default settings are fine for this tutorial, just click on **Launch Server**. You will see
```

File: docs/user/firststeps.rst [hunk 4]
```diff
@@ -29,36 +30,58 @@ On renkulab.io_ you can sign in using with your GitHub
...
+a new row with a yellow button on the right. Wait until its color turns
```
Comment (ableuler) [inline docs/user/firststeps.rst]:
```suggestion
a new row with a yellow button on the right. Wait until its color has turned
```

File: docs/user/firststeps.rst [hunk 5]
```diff
@@ -29,36 +30,58 @@ On renkulab.io_ you can sign in using with your GitHub
...
+to green and the text from **Starting** to **Running**
```
Comment (ableuler) [inline docs/user/firststeps.rst]:
```suggestion
to green and the text from **Starting** to **Running**.
```

File: docs/user/firststeps.rst [hunk 6]
```diff
@@ -29,36 +30,58 @@ On renkulab.io_ you can sign in using with your GitHub
...
-Add input data to your project
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Adding data to your project
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 In the JupyterLab interface, we can see that a few files already exist.
 Let's start by adding data using the `Renku CLI <https://renku-python.readthedocs.io/en/latest/cli.html>`__.
 
-From JupyterLab, start a terminal.
+From JupyterLab, start a terminal by clicking in the **Launcher** page on
```
Comment (ableuler) [inline docs/user/firststeps.rst]: I suggest:
```
From JupyterLab, start a terminal by clicking the **Terminal** icon (1) on the bottom right of the **Launcher** page.
```

File: docs/user/firststeps.rst [hunk 7]
```diff
@@ -110,40 +133,59 @@ repository:
...
+    # Locking support detected on remote "origin". Consider enabling it with: [...]
+    # Counting objects: 19, done. (1/1), 66 MB | 0 B/s
+    # Delta compression using up to 8 threads.
+    # Compressing objects: 100% (15/15), done.
+    # Writing objects: 100% (19/19), 2.26 KiB | 463.00 KiB/s, done.
+    # Total 19 (delta 3), reused 0 (delta 0)
+    # To https://renkulab.io/gitlab/john.doe/zurich-bikes-tutorial.git
+    #     b55aea9..91b226b  master --> master
+
+The data file can be opened from JupyterLab by going to the **File** tab
```
Comment (ableuler) [inline docs/user/firststeps.rst]:
```suggestion
The data file can be opened from JupyterLab by browsing to the **File** tab
```

File: docs/user/firststeps.rst [hunk 8]
```diff
@@ -110,40 +133,59 @@ repository:
...
+on the top left (1) and traversing the ``data`` folder (2).
```
Comment (ableuler) [inline docs/user/firststeps.rst]:
```suggestion
on the top left (1) and clicking the ``data`` folder (2).
```

File: docs/user/firststeps.rst [hunk 9]
```diff
@@ -29,36 +30,58 @@ On renkulab.io_ you can sign in using with your GitHub
...
+.. image:: ../_static/images/ui_connect_to_server.png
+    :width: 100%
+    :align: center
+    :alt: Start a new server
 
 .. _add_data:
 
-Add input data to your project
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Adding data to your project
```
Comment (ableuler) [inline docs/user/firststeps.rst]:
```suggestion
Add data to your project
```

File: docs/user/firststeps.rst [hunk 10]
```diff
@@ -153,19 +195,20 @@ Opening the file, we can see it contains some data in CSV format.
 
 .. _python_environment:
 
-Install and manage Python packages
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Installing and managing Python packages
```
Comment (ableuler) [inline docs/user/firststeps.rst]: I'd stick to the original formulation here.

File: docs/user/firststeps.rst [hunk 11]
```diff
@@ -252,10 +335,10 @@ and type ``git push`` to sync with the server.
 
 .. _interactive_exploration:
 
-Interactively explore the bicycle data
-""""""""""""""""""""""""""""""""""""""
+Interactively exploring the bicycle counting data
```
Comment (ableuler) [inline docs/user/firststeps.rst]:
```suggestion
Interactively explore the bicycle count data
```

File: docs/user/firststeps.rst [hunk 12]
```diff
@@ -196,8 +269,8 @@ pre-installed in the new server's environment.
 
 .. _jupyterlab:
 
-Work using JupyterLab
-^^^^^^^^^^^^^^^^^^^^^
+Working with Renku within JupyterLab
```
Comment (ableuler) [inline docs/user/firststeps.rst]:
```suggestion
Use Renku from within JupyterLab
```

File: docs/user/firststeps.rst [hunk 13]
```diff
@@ -304,12 +408,16 @@ When you are satisfied with the code you can commit it to your repository:
...
+    # [...]
+    # To https://dev.renku.ch/gitlab/john.doe/zurich-bikes-tutorial.git
+    #     a40f192..7922ee1  master -> master
 
 
 .. _create_workflow:
 
-Produce a repeatable workflow
-"""""""""""""""""""""""""""""
+Producing a repeatable analysis
```
Comment (ableuler) [inline docs/user/firststeps.rst]: I'd stick to the original here

File: docs/user/firststeps.rst [hunk 14]
```diff
@@ -352,17 +475,49 @@ changes to the repository. See the `renku command line docs <https://renku-
...
+    If you played around with the tutorial and you started a new
+    Jupyter server, this may happen. Why?
+    `Under the hood <https://renku-python.readthedocs.io/en/latest/api.html>`_,
+    we use
+    `git-lfs <https://git-lfs.github.com/>`_
+    to save big files and we automatically fetch them only
+    when it is required by a `renku` command. If you check the
+    ``2019_verkehrszaehlungen_werte_fussgaenger_velo.csv`` file you
+    will see only a few lines of metadata starting with
+    ``version https:://fit-lfs.github.com/spec/v1``. You can easily
```
Comment (ableuler) [inline docs/user/firststeps.rst]:
```suggestion
    ``version https://git-lfs.github.com/spec/v1``. You can easily
```

File: docs/user/firststeps.rst [hunk 15]
```diff
@@ -352,17 +475,49 @@ changes to the repository. See the `renku command line docs <https://renku-
...
+    fetch the data manually by pulling from the console
```
Comment (ableuler) [inline docs/user/firststeps.rst]:
```suggestion
    fetch the data manually from the console by running 
```

File: docs/user/firststeps.rst [hunk 16]
```diff
@@ -420,12 +602,21 @@ Lastly, let's not forget to push our work:
...
+    # Total 39 (delta 14), reused 0 (delta 0)
+    # To https://dev.renku.ch/gitlab/lorenzo.cavazzi.tech/deleteme.git
+    #    8892173..8d00b71  master -> master
 
 
 .. _sharing_is_caring:
 
-Share your results and collaborate with your peers
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Sharing your results and collaborating with your peers
```
Comment (ableuler) [inline docs/user/firststeps.rst]: Again, I'd revert to the original formulation.

**Comment by ableuler** [comment]
Tanks for this update @lorenzo-cavazzi ! 👍

File: docs/user/firststeps.rst (hunk 8 above)
Comment (lorenzo-cavazzi) [reply docs/user/firststeps.rst]: changed slightly since the file is in a sub folder

Comment (ableuler) [reply docs/user/firststeps.rst]: ok

**Pull request merged**